QModManager/API/ModLoading/IQMod.cs

```diff
@@ -0,0 +1,80 @@
+﻿namespace QModManager.API.ModLoading
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+
+    /// <summary>
+    /// All public data about a QMod.
+    /// </summary>
+    public interface IQMod
+    {
+        /// <summary>
+        /// The ID of the mod <para/>
+        /// Can only contain alphanumeric characters and underscores: (<see langword="a-z"/>, <see langword="A-Z"/>, <see langword="0-9"/>, <see langword="_"/>)
+        /// </summary>
+        string Id { get; }
+
+        /// <summary>
+        /// The display name of the mod
+        /// </summary>
+        string DisplayName { get; }
+
+        /// <summary>
+        /// The author of the mod
+        /// </summary>
+        string Author { get; }
+
+        /// <summary>
+        /// The game this mod was developed for.
+        /// </summary>
+        Game ParsedGame { get; }
+
+        /// <summary>
+        /// The dependencies of the mod and their optional minimum required version
+        /// </summary>
+        Dictionary<string, Version> RequiredMods { get; }
+
+        /// <summary>
+        /// A list of mods, before which, this mod will load
+        /// </summary>
+        string[] LoadBefore { get; }
+
+        /// <summary>
+        /// A list of mods, after which, this mod will load
+        /// </summary>
+        string[] LoadAfter { get; }
```

Comment:
Ideally, this would be some `readonly` type of collection.  
This possible now since `IQModSerializable` enforces the requirements for the old json files.

Reply:
Note that `.NET 3.5` does have `IReadOnlyCollection` or `IReadOnlyList`.  
So the best we can do here is an `IEnumerable`.